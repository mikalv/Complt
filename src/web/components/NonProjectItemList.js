import React from 'react';
import { route } from 'preact-router';
import ItemList from './ItemList';

const NonProjectItemList = props => (
  <ItemList
    className="flex-child"
    onLeftButtonClick={i =>
      props.completeItem(props.items[i]._id, !props.items[i].isCompleted)}
    onItemUpdate={i => props.showUpdateItemDialog(props.items[i]._id)}
    items={props.items}
    onItemTap={i => {
      if (props.items[i].isProject) {
        route(`/project/${props.items[i]._id}`);
      }
    }}
    canMove
    canDeleteTask
    canDeleteProject
    onDelete={i => {
      const item = props.items[i];
      if (item.parent && item.parent._id) {
        props.deleteItem(item.parent._id, item._id);
      } else props.deleteItemWithoutParent(item._id);
    }}
    onItemMove={i => {
      const item = props.items[i];
      if (item.parent && item.parent._id) {
        props.showMoveItemDialog(item._id, item.parent._id);
      } else {
        props.showMoveItemDialog(item._id, null);
      }
    }}
  />
);

export default NonProjectItemList;
