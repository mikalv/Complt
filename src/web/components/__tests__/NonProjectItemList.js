import React from 'react';
import { mount } from 'enzyme';
import NonProjectItemList from '../NonProjectItemList';

const task = { name: 'Task', isProject: false, isCompleted: false, tags: ['@tag'] };
const project = { name: 'Project', isProject: true };

const items = [
  { ...task, _id: 'item1' },
  { ...project, _id: 'item2' },
  { ...task, _id: 'item3' },
];

describe('NonProjectItemList component', () => {
  it('renders without crashing', () => {
    mount(<NonProjectItemList />);
  });
  it('renders correctly', () => {
    const component = mount(<NonProjectItemList items={items} />);
    expect(component).toMatchSnapshot();
  });
  it('calls completeItem correctly when an item is completed', () => {
    const completeItem = jest.fn();
    const component = mount(<NonProjectItemList
      items={items}
      completeItem={completeItem}
    />);
    component.childAt(2).find('Avatar').simulate('click');
    expect(completeItem).toBeCalledWith('item3', true);
  });
  it('calls routerPush correctly when a project is clicked', () => {
    const routerPush = jest.fn();
    const component = mount(<NonProjectItemList
      items={items}
      history={{ push: routerPush }}
    />);
    component.childAt(1).find('li').find('AccessibleFakeButtonInked')
    .simulate('click');
    expect(routerPush).toBeCalledWith('/project/item2');
  });
  it('calls showUpdateItemDialog when an item is updated', () => {
    const showUpdateItemDialog = jest.fn();
    const component = mount(<NonProjectItemList
      items={items}
      showUpdateItemDialog={showUpdateItemDialog}
    />);
    component.childAt(2).find('MdCreate').simulate('click');
    expect(showUpdateItemDialog).toBeCalledWith('item3');
  });
});
