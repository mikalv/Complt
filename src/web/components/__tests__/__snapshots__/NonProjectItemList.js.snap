exports[`NonProjectItemList component renders correctly 1`] = `
<Component
  items={
    Array [
      Object {
        "_id": "item1",
        "isCompleted": false,
        "isProject": false,
        "name": "Task",
        "tags": Array [
          "@tag",
        ],
      },
      Object {
        "_id": "item2",
        "isProject": true,
        "name": "Project",
      },
      Object {
        "_id": "item3",
        "isCompleted": false,
        "isProject": false,
        "name": "Task",
        "tags": Array [
          "@tag",
        ],
      },
    ]
  }>
  <Component
    items={
      Array [
        Object {
          "_id": "item1",
          "isCompleted": false,
          "isProject": false,
          "name": "Task",
          "tags": Array [
            "@tag",
          ],
        },
        Object {
          "_id": "item2",
          "isProject": true,
          "name": "Project",
        },
        Object {
          "_id": "item3",
          "isCompleted": false,
          "isProject": false,
          "name": "Task",
          "tags": Array [
            "@tag",
          ],
        },
      ]
    }
    onItemTap={[Function]}
    onItemUpdate={[Function]}
    onLeftButtonClick={[Function]}>
    <div>
      <Component
        item={
          Object {
            "_id": "item1",
            "isCompleted": false,
            "isProject": false,
            "name": "Task",
            "tags": Array [
              "@tag",
            ],
          }
        }
        onDelete={[Function]}
        onItemTap={[Function]}
        onItemUpdate={[Function]}
        onLeftButtonClick={[Function]}>
        <AccessibleFakeButtonInked
          className="flex row Item"
          component="div"
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          onClick={[Function]}>
          <AccessibleFakeButton
            className="flex row Item"
            component="div"
            ink={
              <InkContainer
                className={undefined}
                disabledInteractions={undefined}
                inkClassName={undefined}
                inkStyle={undefined}
                style={undefined}
                transitionEnterTimeout={450}
                transitionLeaveTimeout={300}
                transitionOverlap={150}
                waitForInkTransition={undefined} />
            }
            onClick={[Function]}
            role="button"
            tabIndex={0}>
            <div
              aria-pressed={false}
              className="md-fake-btn md-pointer--hover flex row Item"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={0}>
              <InkContainer
                transitionEnterTimeout={450}
                transitionLeaveTimeout={300}
                transitionOverlap={150}>
                <ReactTransitionGroup
                  childFactory={[Function]}
                  className="md-ink-container"
                  component="div">
                  <div
                    className="md-ink-container" />
                </ReactTransitionGroup>
              </InkContainer>
              <div
                className="Item-left">
                <ButtonTooltipedInked
                  icon={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}>
                  <ButtonTooltiped
                    icon={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    tooltipDelay={0}
                    tooltipPosition="bottom"
                    tooltipTransitionEnterTimeout={150}
                    tooltipTransitionLeaveTimeout={150}>
                    <Button
                      fixedPosition="br"
                      icon={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined} />
                      }
                      onClick={[Function]}
                      type="button">
                      <button
                        className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button">
                        <InkContainer
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}>
                          <ReactTransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div">
                            <div
                              className="md-ink-container" />
                          </ReactTransitionGroup>
                        </InkContainer>
                        <FontIcon
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons">
                            <MdDone>
                              <IconBase
                                viewBox="0 0 40 40">
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": undefined,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                  width="1em">
                                  <g>
                                    <path
                                      d="m15 27l17.7-17.7 2.3 2.3-20 20-9.3-9.3 2.3-2.3z" />
                                  </g>
                                </svg>
                              </IconBase>
                            </MdDone>
                          </i>
                        </FontIcon>
                      </button>
                    </Button>
                  </ButtonTooltiped>
                </ButtonTooltipedInked>
              </div>
              <div
                className="Item-center">
                <span
                  className="Item-name">
                  Task
                </span>
                <div
                  className="Item-chip-container">
                  <a
                    className="Item-chip-link"
                    onClick={[Function]}
                    to="/tag/@tag">
                    <Chip
                      className="Item-chip"
                      label="@tag"
                      rotateIcon={true}>
                      <button
                        className="md-chip Item-chip"
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        type="button">
                        <span
                          className="md-chip-text">
                          @tag
                        </span>
                      </button>
                    </Chip>
                  </a>
                </div>
              </div>
              <div
                className="Item-right">
                <ButtonTooltipedInked
                  icon={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}>
                  <ButtonTooltiped
                    icon={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    tooltipDelay={0}
                    tooltipPosition="bottom"
                    tooltipTransitionEnterTimeout={150}
                    tooltipTransitionLeaveTimeout={150}>
                    <Button
                      fixedPosition="br"
                      icon={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined} />
                      }
                      onClick={[Function]}
                      type="button">
                      <button
                        className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button">
                        <InkContainer
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}>
                          <ReactTransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div">
                            <div
                              className="md-ink-container" />
                          </ReactTransitionGroup>
                        </InkContainer>
                        <FontIcon
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons">
                            <MdCreate>
                              <IconBase
                                viewBox="0 0 40 40">
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": undefined,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                  width="1em">
                                  <g>
                                    <path
                                      d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z" />
                                  </g>
                                </svg>
                              </IconBase>
                            </MdCreate>
                          </i>
                        </FontIcon>
                      </button>
                    </Button>
                  </ButtonTooltiped>
                </ButtonTooltipedInked>
              </div>
              <hr />
            </div>
          </AccessibleFakeButton>
        </AccessibleFakeButtonInked>
      </Component>
      <Divider>
        <hr
          className="md-divider" />
      </Divider>
      <Component
        item={
          Object {
            "_id": "item2",
            "isProject": true,
            "name": "Project",
          }
        }
        onDelete={[Function]}
        onItemTap={[Function]}
        onItemUpdate={[Function]}
        onLeftButtonClick={[Function]}>
        <AccessibleFakeButtonInked
          className="flex row Item"
          component="a"
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          onClick={[Function]}
          to="project/item2">
          <AccessibleFakeButton
            className="flex row Item"
            component="a"
            ink={
              <InkContainer
                className={undefined}
                disabledInteractions={undefined}
                inkClassName={undefined}
                inkStyle={undefined}
                style={undefined}
                transitionEnterTimeout={450}
                transitionLeaveTimeout={300}
                transitionOverlap={150}
                waitForInkTransition={undefined} />
            }
            onClick={[Function]}
            role="button"
            tabIndex={0}
            to="project/item2">
            <a
              aria-pressed={false}
              className="md-fake-btn md-pointer--hover flex row Item"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={0}
              to="project/item2">
              <InkContainer
                transitionEnterTimeout={450}
                transitionLeaveTimeout={300}
                transitionOverlap={150}>
                <ReactTransitionGroup
                  childFactory={[Function]}
                  className="md-ink-container"
                  component="div">
                  <div
                    className="md-ink-container" />
                </ReactTransitionGroup>
              </InkContainer>
              <div
                className="Item-left">
                <ButtonTooltipedInked
                  icon={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}>
                  <ButtonTooltiped
                    icon={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    tooltipDelay={0}
                    tooltipPosition="bottom"
                    tooltipTransitionEnterTimeout={150}
                    tooltipTransitionLeaveTimeout={150}>
                    <Button
                      fixedPosition="br"
                      icon={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined} />
                      }
                      onClick={[Function]}
                      type="button">
                      <button
                        className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button">
                        <InkContainer
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}>
                          <ReactTransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div">
                            <div
                              className="md-ink-container" />
                          </ReactTransitionGroup>
                        </InkContainer>
                        <FontIcon
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons">
                            <MdAssignment>
                              <IconBase
                                viewBox="0 0 40 40">
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": undefined,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                  width="1em">
                                  <g>
                                    <path
                                      d="m28.4 15v-3.4h-16.8v3.4h16.8z m0 6.6v-3.2h-16.8v3.2h16.8z m-5 6.8v-3.4h-11.8v3.4h11.8z m-3.4-23.4c-0.9 0-1.6 0.7-1.6 1.6s0.7 1.8 1.6 1.8 1.6-0.8 1.6-1.8-0.7-1.6-1.6-1.6z m11.6 0c1.8 0 3.4 1.6 3.4 3.4v23.2c0 1.8-1.6 3.4-3.4 3.4h-23.2c-1.8 0-3.4-1.6-3.4-3.4v-23.2c0-1.8 1.6-3.4 3.4-3.4h6.9c0.7-2 2.5-3.4 4.7-3.4s4 1.4 4.7 3.4h6.9z" />
                                  </g>
                                </svg>
                              </IconBase>
                            </MdAssignment>
                          </i>
                        </FontIcon>
                      </button>
                    </Button>
                  </ButtonTooltiped>
                </ButtonTooltipedInked>
              </div>
              <div
                className="Item-center">
                <span
                  className="Item-name">
                  Project
                </span>
              </div>
              <div
                className="Item-right">
                <ButtonTooltipedInked
                  icon={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}>
                  <ButtonTooltiped
                    icon={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    tooltipDelay={0}
                    tooltipPosition="bottom"
                    tooltipTransitionEnterTimeout={150}
                    tooltipTransitionLeaveTimeout={150}>
                    <Button
                      fixedPosition="br"
                      icon={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined} />
                      }
                      onClick={[Function]}
                      type="button">
                      <button
                        className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button">
                        <InkContainer
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}>
                          <ReactTransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div">
                            <div
                              className="md-ink-container" />
                          </ReactTransitionGroup>
                        </InkContainer>
                        <FontIcon
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons">
                            <MdCreate>
                              <IconBase
                                viewBox="0 0 40 40">
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": undefined,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                  width="1em">
                                  <g>
                                    <path
                                      d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z" />
                                  </g>
                                </svg>
                              </IconBase>
                            </MdCreate>
                          </i>
                        </FontIcon>
                      </button>
                    </Button>
                  </ButtonTooltiped>
                </ButtonTooltipedInked>
              </div>
              <hr />
            </a>
          </AccessibleFakeButton>
        </AccessibleFakeButtonInked>
      </Component>
      <Divider>
        <hr
          className="md-divider" />
      </Divider>
      <Component
        item={
          Object {
            "_id": "item3",
            "isCompleted": false,
            "isProject": false,
            "name": "Task",
            "tags": Array [
              "@tag",
            ],
          }
        }
        onDelete={[Function]}
        onItemTap={[Function]}
        onItemUpdate={[Function]}
        onLeftButtonClick={[Function]}>
        <AccessibleFakeButtonInked
          className="flex row Item"
          component="div"
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          onClick={[Function]}>
          <AccessibleFakeButton
            className="flex row Item"
            component="div"
            ink={
              <InkContainer
                className={undefined}
                disabledInteractions={undefined}
                inkClassName={undefined}
                inkStyle={undefined}
                style={undefined}
                transitionEnterTimeout={450}
                transitionLeaveTimeout={300}
                transitionOverlap={150}
                waitForInkTransition={undefined} />
            }
            onClick={[Function]}
            role="button"
            tabIndex={0}>
            <div
              aria-pressed={false}
              className="md-fake-btn md-pointer--hover flex row Item"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={0}>
              <InkContainer
                transitionEnterTimeout={450}
                transitionLeaveTimeout={300}
                transitionOverlap={150}>
                <ReactTransitionGroup
                  childFactory={[Function]}
                  className="md-ink-container"
                  component="div">
                  <div
                    className="md-ink-container" />
                </ReactTransitionGroup>
              </InkContainer>
              <div
                className="Item-left">
                <ButtonTooltipedInked
                  icon={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}>
                  <ButtonTooltiped
                    icon={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    tooltipDelay={0}
                    tooltipPosition="bottom"
                    tooltipTransitionEnterTimeout={150}
                    tooltipTransitionLeaveTimeout={150}>
                    <Button
                      fixedPosition="br"
                      icon={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined} />
                      }
                      onClick={[Function]}
                      type="button">
                      <button
                        className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button">
                        <InkContainer
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}>
                          <ReactTransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div">
                            <div
                              className="md-ink-container" />
                          </ReactTransitionGroup>
                        </InkContainer>
                        <FontIcon
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons">
                            <MdDone>
                              <IconBase
                                viewBox="0 0 40 40">
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": undefined,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                  width="1em">
                                  <g>
                                    <path
                                      d="m15 27l17.7-17.7 2.3 2.3-20 20-9.3-9.3 2.3-2.3z" />
                                  </g>
                                </svg>
                              </IconBase>
                            </MdDone>
                          </i>
                        </FontIcon>
                      </button>
                    </Button>
                  </ButtonTooltiped>
                </ButtonTooltipedInked>
              </div>
              <div
                className="Item-center">
                <span
                  className="Item-name">
                  Task
                </span>
                <div
                  className="Item-chip-container">
                  <a
                    className="Item-chip-link"
                    onClick={[Function]}
                    to="/tag/@tag">
                    <Chip
                      className="Item-chip"
                      label="@tag"
                      rotateIcon={true}>
                      <button
                        className="md-chip Item-chip"
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        type="button">
                        <span
                          className="md-chip-text">
                          @tag
                        </span>
                      </button>
                    </Chip>
                  </a>
                </div>
              </div>
              <div
                className="Item-right">
                <ButtonTooltipedInked
                  icon={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}>
                  <ButtonTooltiped
                    icon={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    tooltipDelay={0}
                    tooltipPosition="bottom"
                    tooltipTransitionEnterTimeout={150}
                    tooltipTransitionLeaveTimeout={150}>
                    <Button
                      fixedPosition="br"
                      icon={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined} />
                      }
                      onClick={[Function]}
                      type="button">
                      <button
                        className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button">
                        <InkContainer
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}>
                          <ReactTransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div">
                            <div
                              className="md-ink-container" />
                          </ReactTransitionGroup>
                        </InkContainer>
                        <FontIcon
                          iconClassName="material-icons">
                          <i
                            className="md-icon material-icons">
                            <MdCreate>
                              <IconBase
                                viewBox="0 0 40 40">
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": undefined,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                  width="1em">
                                  <g>
                                    <path
                                      d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z" />
                                  </g>
                                </svg>
                              </IconBase>
                            </MdCreate>
                          </i>
                        </FontIcon>
                      </button>
                    </Button>
                  </ButtonTooltiped>
                </ButtonTooltipedInked>
              </div>
              <hr />
            </div>
          </AccessibleFakeButton>
        </AccessibleFakeButtonInked>
      </Component>
      <Divider>
        <hr
          className="md-divider" />
      </Divider>
    </div>
  </Component>
</Component>
`;
